contracts/OctobayGovernor.sol

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.0;
+pragma experimental ABIEncoderV2;
+
+import './OctobayStorage.sol';
+import './OctobayGovTokenFactory.sol';
+import './OctobayGovToken.sol';
+
+// This contract acts as Octobay's storage of all Governors which are used for voting on proposals
+contract OctobayGovernor is OctobayStorage {
+
+    struct Governor {
+        bool isValue; // Ensure we have a valid value in the map
+        uint256 proposalCount; // Number of proposals
+        uint16 newProposalReq; // min percentage required for a token holder to create a new proposal
+        mapping (uint => Proposal) proposalList; // List of proposals
+    }
+
+    struct Proposal {
+        bool isValue; // Ensure we have a valid value in the map
+        address creator; // Creator of this proposal
+        string discussionId; // GitHub Graph ID
+        uint256 startDate; // timestamp from when proposal is Active
+        uint256 endDate; // timestamp for when voting closes on proposal, can be 0 (open ended)
+        uint16 quorum; // min percentage (0 - 10000)
+        int16 voteCount; // the current vote count as a percent of supply
+        uint256 snapshotId; // ID of the snapshot of balances for the token, taken at proposal creation
+        OctobayGovToken votingToken; // governance token required to vote
+        mapping (address => Vote) votesBySubmitter; // map of votes submitted for proposal by submitter
+    }
+
+    struct Vote {
+        bool hasVoted; // to track whether a token holder has already voted
+        int16 vote; // can be negative, represents token share (0 - 10000)
+    }
+
+    enum ProposalState {
+        Pending,
+        Active,
+        Defeated,
+        Succeeded
+    }
+
+    event ProposalCreated(string _projectId, string _discussionId, uint256 _startDate, uint256 _endDate, uint16 _quorum, address _creator, uint256 _proposalId);
```

Comment:
Btw, event parameters don't really need this "_" prefix. (lol... "_" looks a bit like frankenstein :D)
When emitting the event, the param names don't need to match those in the event's definition. So you can keep it short here.

Reply:
Sure, will update